lib/hammer_cli/output/definition.rb

```diff
@@ -8,14 +8,72 @@ def initialize
       @fields = []
     end
 
+    def build(dsl = Dsl.new, &block)
+      dsl.build(&block) if block_given?
+      append(dsl.fields)
+    end
+
     def append(fields)
+      return @fields if fields.nil?
+      fields = [fields] unless fields.is_a? Array
       @fields += fields
     end
 
+    def find_field(field_id)
+      @fields[field_index(field_id)]
+    end
+
+    def insert(mode = :after, field_id = nil, fields = nil, &block)
+      index = field_index(field_id)
+      index += 1 if mode == :after
+      definition = self.class.new
+      definition.append(fields)
+      definition.build(&block)
```

Comment:
This seems too complicated to me. I guess the only reason why you created new definition is the fact that `build` automatically appends the fields. How about moving injection of `Dsl.new` into constructor and using just `@dsl.build` here?